### PR TITLE
Support non-default schemas

### PR DESCRIPTION
**Pull request by suzinyou**
Reviewer: @poornimaramesh 

Estimate: 1 hour

---

### Goal

Add support for tables in non-default schemas

### Changes

Users can now add table description with table name `{schema_name}.{table_name}` for tables in non-default schemas.

When we do DB reflection to get metadata, we look for the presence of `{schema_name}.` prefix and look for tables accordingly. If the schema is unspecified, AAM will look for tables in the default schema.

## How has this been tested?

Connect to a DB with non-default schemas, and make queries.

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [ ] I have resolved merge conflicts
- [ ] I have updated the automated tests (if applicable)
- [ ] I have updated the requirements (if applicable)
- [ ] I have updated the README file (if applicable)
- [ ] I have updated affected documentation (if applicable)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,6 +17,7 @@ pymysql = "^1.1.1"
 aiomysql = "^0.2.0"
 openpyxl = "^3.1.5"
 pandas = "^2.2.3"
+asyncpg = "^0.30.0"
```
Comment (suzinyou) [inline pyproject.toml:20]: Required for making postgres connection in jupyter notebook. Should I move it to dev dependencies?

File: askametric/query_processor/tools.py [hunk 2]
```diff
@@ -59,34 +61,52 @@ async def _get_table_schema(
...
+        def _do_reflect(_: Any) -> defaultdict:
             """Reflect the tables."""
             engine = asession.get_bind()
             inspector = inspect(engine)
-            existing_tables = [
-                table for table in table_list if inspector.has_table(table)
-            ]
-            metadata.reflect(bind=engine, only=existing_tables, views=True)
-            return existing_tables
+            existing_schemas_and_tables = defaultdict(list)
```
Comment (suzinyou) [inline askametric/query_processor/tools.py:68]: Now we're saving schemas in a dict of lists, so

```python
{
    None: ["tables_in_default_schema", "another_table_in_default_schema",],
    "reports": ["table_in_reports_schema", "another_table_in_report"schema",],
}
```

**Review by poornimaramesh — APPROVED** [review]
Works, approving, thanks so much!
Might have to resolve merge conflicts, since Zia's PR modifies similar files / functions.

Some notes for future tickets (nothing urgent):
- need to check if this works for other db flavours also
- this should in principle work for many schemas in the table, but I wonder if we'll run into multiple-join problems more quickly because by nescessity the table description becomes longer / more complicated. Also, it will quickly get tedious to write descriptions for lots of tables per schema. Might be worth thinking about writing a more robust pipeline that can just deal with recursive db structure, and only requires a description for the topmost level of recursion, and everything else is optional?

File: pyproject.toml (hunk 1 above)
Comment (poornimaramesh) [reply pyproject.toml]: No, I think it's ok to leave it in the main dependencies. Anyone trying to connect a postgres db to test will need it...

**Pull request merged**